test/Tests/src/Semantic/Java_Interop_Spec.enso

```diff
@@ -66,4 +68,27 @@ spec =
             april1st.month.should_equal 4
             april1st.day.should_equal 1
 
+    Test.group "Java case of" <|
+        Test.specify "case on Thread.State enum" <|
+            match x = case x of
+                State.NEW -> "new"
+                _ -> "unknown"
+            match State.NEW . should_equal "new"
+            match State.BLOCKED . should_equal "unknown"
+
+        Test.specify "case on String static field" <|
+            match x = case x of
+                String.CASE_INSENSITIVE_ORDER -> "match"
```

Comment:
Shall this be allowed or not? `java.lang.String` isn't enum and `CASE_INSENSITIVE_ORDER` isn't an enum instance. But the current implementation doesn't need to work on `java.lang.Enum`. It works on any Java class with static fields. I could introduce an additional check and refuse anything that isn't an `Enum` subclass.

Let's vote! +1 for allowing access to all static fields of all classes. -1 to allow access only to `Enum` and its values.